mailgun/route_handlers.py

```diff
@@ -54,173 +54,175 @@ def handle_mailing_list_email_route(request):
-                           u'but we\'ve already handled that.  Dropping.',
-                           message_id)
-            return JsonResponse({'success': True})
-
     # if we want to check email addresses against the sender, we need to parse
     # out just the address.
     parsed_sender = address.parse(sender)
     sender_address = parsed_sender.address.lower()
-    sender_display_name = parsed_sender.display_name
-
-    # make sure the mailing list exists
-    bounce_back_email_template = None
-    try:
-        ml = MailingList.objects.get_or_create_or_delete_mailing_list_by_address(recipient)
-    except MailingList.DoesNotExist:
-        logger.info(
-            u'Sending mailing list bounce back email to %s for mailing list %s '
-            u'because the mailing list does not exist', sender, recipient)
-        bounce_back_email_template = get_template('mailgun/email/bounce_back_does_not_exist.html')
-        content = bounce_back_email_template.render(Context({
-            'sender': sender,
-            'recipient': recipient,
-            'subject': subject,
-            'message_body': body_plain or body_html,
-        }))
-        listserv_client.send_mail(recipient, recipient, sender_address,
-                                  subject='Undeliverable mail', html=content,
-                                  message_id=message_id)
-        return JsonResponse({'success': True})
 
-    # Always include teaching staff addresses with members addresses, so that they can email any list in the course
-    teaching_staff_addresses = ml.teaching_staff_addresses
-    member_addresses = teaching_staff_addresses.union([m['address'] for m in ml.members])
-    if ml.access_level == MailingList.ACCESS_LEVEL_MEMBERS and sender_address not in member_addresses:
-        logger.info(
-            u'Sending mailing list bounce back email to %s for mailing list %s '
-            u'because the sender was not a member', sender, recipient)
-        bounce_back_email_template = get_template('mailgun/email/bounce_back_access_denied.html')
-    elif ml.access_level == MailingList.ACCESS_LEVEL_STAFF and sender_address not in teaching_staff_addresses:
-        logger.info(
-            u'Sending mailing list bounce back email to %s for mailing list %s '
-            u'because the sender was not a staff member', sender, recipient)
-        bounce_back_email_template = get_template('mailgun/email/bounce_back_access_denied.html')
-    elif ml.access_level == MailingList.ACCESS_LEVEL_READONLY:
-        logger.info(
-            u'Sending mailing list bounce back email to %s for mailing list %s '
-            u'because the list is readonly', sender, recipient)
-        bounce_back_email_template = get_template('mailgun/email/bounce_back_readonly_list.html')
+    for recipient_address in recipients:
+        recipient = recipient_address.address
+        sender_display_name = parsed_sender.display_name
```

Comment:
No, because we modify sender_display_name down below (around 180 or so). We could modify that logic down below to make it a little cleaner, but I'd rather leave it like this now to limit the changes.